src/com/amazon/ionelement/api/Element.kt

```diff
@@ -0,0 +1,257 @@
+package com.amazon.ionelement.api
+
+import com.amazon.ion.Decimal
+import com.amazon.ion.IntegerSize
+import com.amazon.ion.IonWriter
+import com.amazon.ion.Timestamp
+import java.math.BigInteger
+
+// QUESTION/TODO:  Verify that we are consistent about using *either* the Kotlin type name or the Ion type name.
+// i.e. Ion uses "float" file Kotlin's equivalent is "Double".  Int vs Long, etc.  I think in some places
+// we use one or the other but the intent is that the only place we use the Ion type name is for members of ElementType.
+// everywhere else should refer to the Kotlin type name.
+// The question here is: is this a good idea or if we should just use the Ion type name everywhere?
+
+/**
+ * TODO:  probably length documentation.
+ *
+ * TODO: when writing documentation, note that every implementation of [Element] must also implement [IonElement].
+ * The reason:  has to do with the need for all constructor functions accept [Element] but for the collection elements
+ * to return instances of [IonElement].
+ *
+ */
+interface Element {
```

Comment:
While discussing this PR with a customer, he found the name `Element` to be a little confusing since in combination with `IonElement` it seems to imply there are elements.

One possible solution to this would be to name `IonElement` to `AnyIonElement` and `Element` to `IonElement`.

This *would* reflect the intended use of these types better, IMHO.